include/drivers/pinctrl.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2021 Piotr Mienkowski
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public APIs for PINCTRL drivers
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_PINCTRL_H_
+#define ZEPHYR_INCLUDE_DRIVERS_PINCTRL_H_
+
+#include <zephyr/types.h>
+#include <toolchain.h>
+#include <soc_pinctrl.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief PINCTRL Driver APIs
+ * @defgroup pinctrl_interface PINCTRL Driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * @typedef pinctrl_soc_pins_t
```

Comment:
The naming here feels confused.

This looks like it should be called pinctrl_dt_spec_t because PINCTRL_DT_SPEC_GET expands to an initializer for this type.

But then the next patch adds struct pinctrl_pins_dt_spec, which contains one of these plus a num_pins.

I think that in general foo_dt_spec should be initialized by FOO_DT_SPEC_GET. So probably PINCTRL_DT_SPEC_GET should be renamed to PINCTRL_SOC_PINS_GET.

Reply:
agreed the names are a bit of a mess and need cleaning up.  Trying to have the drivers see SPEC similar to what we have for GPIO, SPI, etc.